internal/configuration/config_test.go

```diff
@@ -0,0 +1,148 @@
+package configuration
+
+import (
+	"github.com/spf13/viper"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestAssembleConfig(t *testing.T) {
+	tests := []struct {
+		name           string
+		configureViper func(v *viper.Viper)
+		want           *Config
+		wantErr        bool
+	}{
+		{
+			name:           "Errors on empty config",
+			configureViper: func(v *viper.Viper) {},
+			want:           nil,
+			wantErr:        true,
+		},
+		{
+			name: "Correctly parses minimal config",
+			configureViper: func(v *viper.Viper) {
+				v.Set("Statuspage.ApiKey", "foo")
+				v.Set("Statuspage.PageID", "bar")
+			},
+			want: &Config{
+				Verbose: false,
+				Client: struct {
+					Redirects int
+					Retries   int
+				}{
+					Redirects: 3,
+					Retries:   3,
+				},
+				Statuspage: struct {
+					ApiKey     string `validate:"required"`
+					PageID     string `validate:"required"`
+					ApiRoot    string
+					Components []Component
+				}{
+					ApiKey:  "foo",
+					PageID:  "bar",
+					ApiRoot: "https://api.statuspage.io/v1",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			v := viper.New()
+			tt.configureViper(v)
+			got, err := AssembleConfig(v)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("AssembleConfig() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("AssembleConfig() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_newDefaultConfig(t *testing.T) {
+	tests := []struct {
+		name string
+		want *Config
+	}{
+		{
+			name: "Default config values",
+			want: &Config{
+				Verbose: false,
+				Client: struct {
+					Redirects int
+					Retries   int
+				}{
+					Redirects: 3,
+					Retries:   3,
+				},
+				Statuspage: struct {
+					ApiKey     string `validate:"required"`
+					PageID     string `validate:"required"`
+					ApiRoot    string
+					Components []Component
+				}{
+					ApiRoot: "https://api.statuspage.io/v1",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := newDefaultConfig(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("newDefaultConfig() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_readEnvironmentVariables(t *testing.T) {
```

Comment:
Ultimately what the read environment Variables method does is just a call to `os.LookupEnv`. IMO we don't need to re-test code from the go standard library or other 3rd party libraries(assuming we've carefully vetted them and they are maintained). We should really just be testing our own logic and functionality. Open to counter arguments on this though.

Reply:
My reasoning for testing it had more to do with "I want to make sure we read the variables we claim to read" so that it is slightly harder for us to accidentally break a deployment that is injecting an env var in the future. If os.LookupEnv fails then the whole test is unstable so hopefully that isn't the case  😅